### ISSUE TITLE
DRYing out main.js - possible solution?

### ISSUE THREAD

**Issue by tacotoemeck** [fac19/week6-hihi#42]
Thinking of "drying" out your main.js , perhaps creating one function such as one below and passing on parameter string as a filter variable could work? 

Not sure about async functionality though but could be worth a try? If that works you'd need to add event listener for each variable you created and pass on different 'filter' parameters to it. 

Lastly, I guess using let might be a better fit than var :dancers: 

```javascript

async function categoryFilter(category) {
  filter = category;
  let newHtml = await fetch("/", {
    method: "POST",
    headers: { "Content-Type": "application/javascript" },
    body: filter,
  });
  document.open("text/html");
  document.write(`${await newHtml.text()}`);
  document.close();
}

workFilter.addEventListener("click", () => categoryFilter("Work"));

**Comment by hannahgooding** [comment]
Agreed! This was something we didn't get round to refactoring from our project last week.  Neither `let` or `const` worked so we had to use `var` but in fact they work without variable keyword assignment! Would love to work out why when we have the chance :)
